enhancements/machine-api/control-plane-machine-set.md

```diff
@@ -0,0 +1,801 @@
+removed by the Machine controller in the normal way.
+
+#### Removing/disabling the ControlPlaneMachineSet
+
+As some users may want to remove or disable the ControlPlaneMachineSet, a finalizer will be placed on the
+`ControlPlaneMachineSet` to allow the controller to ensure a safe removal of the ControlPlaneMachineSet, while leaving
+the Control Plane Machines in place.
+
+Notably it will need to ensure that there are no owner references on the Machines pointing to the ControlPlaneMachineSet
+instances. This will prevent the garbage collector from removing the Control Plane Machines when the
+`ControlPlaneMachineSet` is deleted.
+
+If users later wish to reinstall the `ControlPlaneMachineSet`, they are free to do so.
+
+#### Installing a ControlPlaneMachineSet within an existing cluster
+
+When adding a ControlPlaneMachineSet to a new cluster, the end user will need to define the `ControlPlaneMachineSet`
+resource by copying the existing Control Plane Machine ProviderSpecs.
+Once this is copied, they should remove the failure domain and add the desired failure domains to the `FailureDomains`
+field within the ControlPlaneMachineSet spec.
+
+To ensure adding a `ControlPlaneMachineSet` to the cluster is safe, we will need to ensure via a validating webhook
+that the replica count defined in the spec is consistent with the actual size of the control plane within the cluster.
+
+If no Control Plane Machines exist, or they are in a non-Running state, the operator will report degraded until this
+issue is resolved. This creates a dependency for the `ControlPlaneMachineSet` operator on the Machine API. It will be
+required to run at a higher run-level than Machine API.
+
+In UPI or misconfigured clusters, a user adding a `ControlPlaneMachineSet` will result in a degraded cluster.
+Users will need to remove the invalid ControlPlaneMachineSet resource to restore their cluster to a healthy state.
+
+We do not recommend that UPI users attempt to adopt their control plane instances into Machine API due to the high
+likelihood that they cannot create an accurate configuration to replicate the original control plane instances. This
+limitation also limits the `ControlPlaneMachineSet` operator to an IPI only operator.
+
+#### Naming of Control Plane Machines owned by the ControlPlaneMachineSet
+
+In an IPI OpenShift cluster, the Control Plane Machines are named after the cluster ID followed by an index, for example
+`my-openshift-cluster-abcde-master-0`. As we will need to create additional Machines during replacement operations, we
+cannot reuse the names of the existing Machines. Instead we will use the `generateName` field within the metadata to
+create a random suffix for the new Machines, replacing the index of current Machines. For example, Machines created
+by a `ControlPlaneMachineSet` may have a name such as `my-openshift-cluster-abcde-master-fghij`.
```

Comment:
can't you generate a new random string to include, but still append the appropriate index suffix to the new machine name?

it may not be strictly critical but it is user-friendly

Reply:
We definitely could do this, though with other Machines we tend not to, mainly because we try to stick to the cattle not pets idea, however, for control plane, I think you're right, it might be nice to see the 0, 1, 2 indexes still, will update